sources/cfsignal/Cargo.toml

```diff
@@ -0,0 +1,27 @@
+[package]
+name = "cfsignal"
+version = "0.0.1"
+authors = ["Mello7tre <mello+github@ankot.org>"]
+license = "Apache-2.0 OR MIT"
+edition = "2018"
+publish = false
+# Don't rebuild crate just because of changes to README.
+exclude = ["README.md"]
+
+[dependencies]
+log = "0.4"
+serde = { version = "1.0.100", features = ["derive"] }
+simplelog = "0.10"
+snafu = { version = "0.6" }
+toml = "0.5.1"
+tokio = { version = "1", features = ["full"] }
+rusoto_core = { version = "0.47.0", default-features = false, features = ["rustls"] }
+rusoto_cloudformation = { version = "0.47.0", default-features = false, features = ["rustls"] }
+ec2_instance_metadata = "0.3"
```

Comment:
We have a existing [`imdsclient` crate](https://github.com/bottlerocket-os/bottlerocket/tree/develop/sources/imdsclient) which handles the differences between IMDSv1 and IMDSv2.  It would be better to use that crate versus taking a dependency on `ec2_instance_metadata`.

That crate doesn't yet have a method to grab the instance ID, but [this method](https://github.com/bottlerocket-os/bottlerocket/blob/44a10648318fbb187cc9f7dda6806b254cc77b97/sources/imdsclient/src/lib.rs#L106-L110) could be copied and the target changed to `meta-data/instance-id`.

Reply:
good to know, i will take a look and make the changes